src/util/simplify_expr.cpp

```diff
@@ -153,6 +158,148 @@ simplify_exprt::simplify_popcount(const popcount_exprt &expr)
   return unchanged(expr);
 }
 
+/// Simplify String.endsWith function when argument are constant
+/// \param expr [in,out]: the expression to simplify
+/// \param ns: namespace
+/// \return: true iff expr was not modified
+static simplify_exprt::resultt<> simplify_string_endswith_func(
+  const function_application_exprt &expr,
+  const namespacet &ns)
+{
+  const function_application_exprt &function_app =
+    to_function_application_expr(expr);
```

Comment:
Cast is now unnecessary (also for `isEmpty()` and `compareTo()`).

Reply:
removed